### PR TITLE
Fix memory leaks and assert failures

### PR DESCRIPTION
**Pull request by tobwithu**
Fixed memory leaks and assert failures which were found during build with Visual Studio 2017 on Windows 7

### CODE REVIEW

File: librapidsvn/src/folder_browser.cpp [hunk 1]
```diff
@@ -182,6 +182,11 @@ struct FolderBrowser::Data
   ~Data()
   {
     DeleteAllItems();
+	  if (singleContext != 0)
+	  {
+		  delete singleContext;
+		  singleContext = 0;
+	  }
   }
```
Comment (XelaRellum) [inline librapidsvn/src/folder_browser.cpp]: This can by accomplished by:
```c++
delete singleContext;
singleContext = 0;
```

Comment (tobwithu) [reply librapidsvn/src/folder_browser.cpp]: I think `if` statement is needed.
when `singleContext == 0`, `delete singleContext` can have problem.
Am I missing something?

Comment (XelaRellum) [reply librapidsvn/src/folder_browser.cpp]: It's perfectly fine to use `delete` on a `nullptr` (see https://en.cppreference.com/w/cpp/language/delete).
But the `singleContext = 0` is needed so a second call does not try to delete the pointer again.

Comment (XelaRellum) [reply librapidsvn/src/folder_browser.cpp]: if the case is the same, the same action is valid (haven't checked with the code).

Comment (XelaRellum) [reply librapidsvn/src/folder_browser.cpp]: See:
https://isocpp.org/wiki/faq/freestore-mgmt#delete-handles-null
https://isocpp.org/wiki/faq/freestore-mgmt#double-delete-disaster

**Pull request merged**